APMrover2/mode_auto.cpp

```diff
@@ -131,6 +131,40 @@ float ModeAuto::get_distance_to_destination() const
     return 0.0f;
 }
 
+// get desired location
+bool ModeAuto::get_desired_location(Location& destination)
+{
+    switch (_submode) {
+    case Auto_WP:
+        if (g2.wp_nav.is_destination_valid()) {
+            destination = g2.wp_nav.get_destination();
+            return true;
+        } else {
+            return false;
+        }
+        break;
+    case Auto_HeadingAndSpeed:
+        // always return true because this is the safer option to allow missions to continue
```

Comment:
This is no good for the mavlink message you're starting to send, 'though - you're not filling in destination!

Reply:
ah, definitely a bug, thanks!